### PR TITLE
Add a test to submit a web extension

### PR DESCRIPTION
**Pull request by justinpotts**
Addresses #820

### CODE REVIEW

**Comment by justinpotts** [comment]
@davehunt @krupa: With extreme pleasure, I declare this PR ready for review! Test passes locally, and all other tests are still passing.

I suspect you're going to comment upon me using the Link Text locator in `base.py`, so let me go ahead an explain my reasoning... If the account has never submitted an Add-On before, the Submit link is the first item in the list. After going through the process once, a Manage Submissions tab appears, dropping the Submit Add-on item to number two. I wanted to make sure we were able to run the tests with an account whether it was the first time, or for the tenth time, so I used Link Text to ensure we're always clicking that.

I'm also happy to report we're using Axl, and the extension generation is working beautifully.

Looking forward to hearing your comments!

File: pages/desktop/submit.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    _upload_addon_locator = (By.ID, "upload-addon")
+    _upload_progress_bar_success_locator = (By.CSS_SELECTOR, "#upload-file>.upload-status>.bar-success")
+    _misc_category_locator = (By.CSS_SELECTOR, ".addon-misc-category.checkbox-choices>li>label")
+    _mpl_license_locator = (By.CSS_SELECTOR, "#id_builtin>li>label")
+    _review_button_locator = (By.CSS_SELECTOR, ".submit-buttons>button")
+    _done_next_steps_locator = (By.CSS_SELECTOR, ".done-next-steps")
+    _continue_to_step_3_locator = (By.ID, "submit-upload-file-finish")
+    _continue_to_step_4_locator = (By.CSS_SELECTOR, ".submission-buttons.addon-submission-field>button")
+    _continue_to_step_5_locator = (By.CSS_SELECTOR, ".edit-media-button.addon-submission-field>button")
+    _continue_to_step_6_locator = (By.CSS_SELECTOR, ".submission-buttons.addon-submission-field>button")
```
Comment (davehunt) [inline pages/desktop/submit.py]: Duplicate of `_continue_to_step_4_locator`

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+axl
```
Comment (davehunt) [inline requirements.txt]: Let's pin the version number, for safety.

File: tests/desktop/test_extensions.py [hunk 3]
```diff
@@ -1,16 +1,39 @@
...
 class TestExtensions:
 
     @pytest.mark.native
     @pytest.mark.nondestructive
+    def test_web_extension_submission(self, base_url, selenium, user):
+        home_page = Home(base_url, selenium)
+        home_page.login(user['email'], user['password'])
+
+        assert home_page.is_the_current_page
+        assert home_page.header.is_user_logged_in
```
Comment (davehunt) [inline tests/desktop/test_extensions.py]: I suspect these two asserts are not needed. We use them too much already. :)

File: tests/desktop/test_extensions.py [hunk 4]
```diff
@@ -1,16 +1,39 @@
...
+
+        submit_page = home_page.header.click_submit_a_new_addon()
+
+        path = generate()
+        submit_page.upload_addon(path)
```
Comment (davehunt) [inline tests/desktop/test_extensions.py]: 👍

File: pages/desktop/base.py [hunk 5]
```diff
@@ -130,6 +130,9 @@ class HeaderRegion(Page):
         _account_dropdown_locator = (By.CSS_SELECTOR, "#aux-nav .account ul")
         _logout_locator = (By.CSS_SELECTOR, "li.nomenu.logout > a")
 
+        _tools_dropdown_locator = (By.CLASS_NAME, "tools")
+        _submit_a_new_addon_locator = (By.LINK_TEXT, "Submit a New Add-on")
```
Comment (davehunt) [inline pages/desktop/base.py]: I'm okay with this for now, but could you raise an issue against addons-server to give this element a class or id that we can use for testing?

**Comment by davehunt** [comment]
Looks great, just a few comments. Let's get @krupa to comment too, though I'm not sure she has GitHub notifications - could you drop her an email or ping her on IRC?

File: pages/desktop/submit.py (hunk 1 above)
Comment (justinpotts) [inline pages/desktop/submit.py]: @davehunt Should we have this as a separate locator though, just to ensure readability so we're not using `_continue_to_step_four_locator` in the `continue_to_step_six` method?

File: pages/desktop/base.py (hunk 5 above)
Comment (justinpotts) [inline pages/desktop/base.py]: @davehunt Filed! https://github.com/mozilla/addons-server/issues/3212

**Comment by krupa** [comment]
I'm delighted that we have coverage for this feature. good job, justin!

**Comment by justinpotts** [comment]
Thanks @krupa!

**Comment by stephendonner** [comment]
This fixed https://github.com/mozilla/Addon-Tests/issues/863

**Comment by davehunt** [comment]
This actually fixed #820 😀

**Pull request merged**